diam/sm/sm.go

```diff
@@ -109,11 +111,47 @@ func New(settings *Settings) *StateMachine {
 		settings.AnswerHandlerProxy = defaultHandlerFuncProxy
 	}
 
+	// Prepare the list of supported apps from the dictionary
+	preparedApps := PrepareSupportedApps(dict.Default)
```

Comment:
I don't know, if we want to change SM that much, I would rather use our own implementation of diameter client instead of SM. [An example](https://github.com/trilogy-group/ccab/pull/6485/commits/972d6f9c98034dfce935df550d2a7e0c4f0d1688).